array_schema_experimental.go

```diff
@@ -0,0 +1,43 @@
+package tiledb
+
+/*
+#include <tiledb/tiledb.h>
+#include <tiledb/tiledb_experimental.h>
+#include <stdlib.h>
+*/
+import "C"
+import (
+	"fmt"
+	"runtime"
+	"time"
+)
+
+// NewArraySchemaAtTime allocates a new ArraySchema at the provided createTime.
+func NewArraySchemaAtTime(tdbCtx *Context, arrayType ArrayType, createTime time.Time) (*ArraySchema, error) {
+	return NewArraySchemaAtTimestamp(tdbCtx, arrayType, uint64(createTime.UnixMilli()))
+}
+
+// NewArraySchemaAtTimestamp allocates a new ArraySchema at the provided timestamp.
+func NewArraySchemaAtTimestamp(tdbCtx *Context, arrayType ArrayType, timestamp uint64) (*ArraySchema, error) {
+	var arraySchemaPtr *C.tiledb_array_schema_t
+	var cTimestamp C.uint64_t = C.uint64_t(timestamp)
+	ret := C.tiledb_array_schema_alloc_at_timestamp(tdbCtx.tiledbContext.Get(), C.tiledb_array_type_t(arrayType),
+		cTimestamp, &arraySchemaPtr)
+	runtime.KeepAlive(tdbCtx)
+	if ret != C.TILEDB_OK {
+		return nil, fmt.Errorf("error creating tiledb arraySchema: %w", tdbCtx.LastError())
+	}
+	return newArraySchemaFromHandle(tdbCtx, newArraySchemaHandle(arraySchemaPtr)), nil
+}
```

Comment:
We said that `tiledb_array_schema_alloc_at_timestamp` will not be exposed in higher-level APIs (TileDB-Inc/TileDB#5484). Does the server need it? I don't think so, it mostly gets schemas by deserializing them.

Reply:
Done. I left `Arrayschema.TimestampRange`, check out the test and let me know if you have a better way to validate it.